tests/testcases/mod.rs

```diff
@@ -635,4 +735,108 @@ pub mod all {
+        );
+        let tc = TestCaseBuilder::default()
+            .name("http-go".to_string())
+            .template(Some("http-go".to_string()))
+            .appname(Some("http-go-registry-generated".to_string()))
+            .push_to_registry(Some(registry_app_url.clone()))
+            .deploy_args(vec![
+                "--from-registry".to_string(),
+                registry_app_url.clone(),
+                "--insecure".to_string(),
+            ])
+            .assertions(
+                |metadata: AppMetadata,
+                 stdout_stream: Option<BufReader<ChildStdout>>,
+                 stderr_stream: Option<BufReader<ChildStderr>>| {
+                    Box::pin(checks(metadata, stdout_stream, stderr_stream))
+                },
+            )
+            .build()
+            .unwrap();
+
+        tc.run(controller).await.unwrap()
+    }
+
+    pub async fn longevity_apps_works(controller: &dyn Controller) {
+        async fn checks(
+            metadata: AppMetadata,
+            _: Option<BufReader<ChildStdout>>,
+            _: Option<BufReader<ChildStderr>>,
+        ) -> Result<()> {
+            assert_http_response(
+                get_url(metadata.base.as_str(), "/golang").as_str(),
+                200,
+                &[],
+                Some("Hello Fermyon!\n"),
+            )
+            .await?;
+
+            assert_http_response(
+                get_url(metadata.base.as_str(), "/rust").as_str(),
+                200,
+                &[],
+                Some("Hello, Fermyon"),
+            )
+            .await?;
+
+            assert_http_response(
+                get_url(metadata.base.as_str(), "/javascript").as_str(),
+                200,
+                &[],
+                Some("Hello from JS-SDK"),
+            )
+            .await?;
+
+            assert_http_response(
+                get_url(metadata.base.as_str(), "/typescript").as_str(),
+                200,
+                &[],
+                Some("Hello from TS-SDK"),
+            )
+            .await?;
+
+            Ok(())
+        }
+
+        let tc = TestCaseBuilder::default()
+            .name("longevity-apps-test".to_string())
+            .appname(Some("longevity-apps-test".to_string()))
+            .assertions(
+                |metadata: AppMetadata,
+                 stdout_stream: Option<BufReader<ChildStdout>>,
+                 stderr_stream: Option<BufReader<ChildStderr>>| {
+                    Box::pin(checks(metadata, stdout_stream, stderr_stream))
+                },
+            )
+            .build()
+            .unwrap();
+
+        let version = match spin::version() {
+            Ok(version) => version,
+            Err(err) => {
+                println!("{}", err);
+                "".to_string()
+            }
+        };
+        let msg = format!("app using a wasm module built by 'spin version 0.9.0 (a99ed51 2023-02-16)' no longer works with {}", version);
```

Comment:
Can we avoid hard-coding the version or perhaps pull it from some canonical source (variable or const?) that is known/expected to be updated if/when the longevity wasm files are built with a different version?  Or, we can iterate/brainstorm a bit on this if preferred -- but minimally for this PR perhaps add a mention in the longevity-apps-test/README.md to update this string on such events.

Reply:
Yes, this is a good call. I don't know if we can fetch that info dynamically, but I have added a const within the test case with that value for now.